Fix db backup failures

DB backups often fail with:

```
Run ls -lh backup.sql
  ls -lh backup.sql
  shell: /usr/bin/bash -e {0}
  env:
    DB_SERVICE: nrrd-a-psql
    CF_SPACE: prod
ls: cannot access 'backup.sql': No such file or directory
Error: Process completed with exit code 2.
```

This is likely due to a silent error in the cf backup itself.

--- a/.github/workflows/db-backup.yml
+++ b/.github/workflows/db-backup.yml
@@ -53,12 +53,19 @@ jobs:
           cf login -u ${{ secrets.CF_USERNAME }} -p ${{ secrets.CF_PASSWORD }} -a api.fr.cloud.gov -o doi-onrr -s "$CF_SPACE"
 
       - name: Backup database
-        working-directory: ${{ github.workspace }}
-        run: |
-          cg-manage-rds export -o "-F p --no-owner --no-acl" -f backup.sql "$DB_SERVICE" 2>&1 | cat
-
-      - name: Verify backup file
-        run: ls -lh backup.sql
+        uses: nick-fields/retry@v3
+        with:
+          timeout_minutes: 30
+          max_attempts: 3
+          retry_wait_seconds: 30
+          shell: bash
+          command: |
+            set -eo pipefail
+            cd "${{ github.workspace }}"
+            rm -f backup.sql
+            cg-manage-rds export -o "-F p --no-owner --no-acl" -f backup.sql "$DB_SERVICE" 2>&1
+            test -s backup.sql
+            ls -lh backup.sql
 
       - name: Compress backup
         run: |

